### PR TITLE
Use the fred Redis client

### LINKED ISSUE #2606: Move Redis TLS to Rustls
Follow up to #2332

Redis TLS support reintroduced openssl in the Router, which we want to avoid for ease of deployment and security management reasons.

Rustls support is in development and should soon be merged: https://github.com/redis-rs/redis-rs/pull/725
Work has not started for redis-cluster-async though

### LINKED ISSUE #2623: Ubuntu: libssl.so.1.1: cannot open shared object file: No such file or directory
This pinned issue shows you how to work around a temporary openssl related issue.
Ubuntu 22.XX doesn't provide libssl@1.1 anymore. The router will fail to start and show this error:

```bash
error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
```

This happens because ubuntu only ships libssl 3.X:

```bash
$ openssl version
OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)
```

A known workaround is to manually install libssl 1.1 as explained in this [stackoverflow issue](https://stackoverflow.com/questions/72133316/libssl-so-1-1-cannot-open-shared-object-file-no-such-file-or-directory) by following the steps outlined in the answer.

We will keep this issue open for discoverability purposes, until we completely move to rustls.

### PR DESCRIPTION
**Pull request by Geal**
Fix #2623
Fix #2606

This replaces our usage of the `redis` and `redis-cluster-async` crates with `fred`, which comes with support for Redis cluster and rustls out of the box. Rustls support is only available in fred 6.0.0-beta.2 though, not in a stable release.

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

### CODE REVIEW

**Comment by abernix** [comment]
@Geal does this just need to be conflict-resolved?

**Comment by Geal** [comment]
yes, I edited it, the CI command I added was not really necessary

File: .changesets/maint_geal_redis_fred.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+### Use the `fred` Redis client ([Issue #2623](https://github.com/apollographql/router/issues/2623))
+
+A description of the fix which stands on its own separate from the title.  It should embrace the use of Markdown to stylize the commentary so it looks great on the GitHub Releases, when shared on social cards, etc.
```
Comment (bnjjj) [inline .changesets/maint_geal_redis_fred.md:3]: Update this boilerplate

**Review by bnjjj — APPROVED** [review]

**Comment by Geal** [comment]
any way I can nudge the CLA bot so it passes?

**Pull request merged**